src/payment/bolt11.rs

```diff
@@ -408,13 +408,25 @@ impl Bolt11Payment {
 			..PaymentDetailsUpdate::new(payment_id)
 		};
 
-		if !self.payment_store.update(&update)? {
-			log_error!(
-				self.logger,
-				"Failed to manually fail unknown payment with hash: {}",
-				payment_hash
-			);
-			return Err(Error::InvalidPaymentHash);
+		match self.payment_store.update(&update) {
+			Ok(true) => (),
+			Ok(false) => {
+				log_error!(
+					self.logger,
+					"Failed to manually fail unknown payment with hash {}",
+					payment_hash,
+				);
+				return Err(Error::InvalidPaymentHash);
+			},
+			Err(e) => {
+				log_error!(
+					self.logger,
+					"Failed to manually fail unknown payment with hash {}: {}",
```

Comment:
Is the only change in this file that the Err case is logged here? That seems to be independent of the changed 'update' semantics?

Reply:
But it is already the case in main that calling fail twice fails, and this commit doesn't change that?